python/ray/dag/collective_node.py

```diff
@@ -124,8 +129,32 @@ def execute(self, send_buf: "torch.Tensor") -> "torch.Tensor":
         if not isinstance(send_buf, torch.Tensor):
             raise ValueError("Expected a torch tensor")
         communicator = self.get_communicator()
-        recv_buf = torch.empty_like(send_buf)
-        communicator.allreduce(send_buf, recv_buf, self._op)
+        if self._op is None:
```

Comment:
This should be the last "else" branch. If it is None, we are doing allgather. Add a comment here.